extensions/amp-story/1.0/amp-story.js

```diff
@@ -819,6 +828,23 @@ export class AmpStory extends AMP.BaseElement {
     return storyLayoutPromise;
   }
 
+  /**
+   * Retrieves the initial pageId to begin the story with. In order, the
+   * initial page for a story should be the page ID in the history, a valid
+   * page ID in the URL fragment, and then the first page of the story.
+   * @param {Element} firstPageEl
+   * @return {string}
+   * @private
+   */
+  getInitialPageId_(firstPageEl) {
+    const initialPageId = this.getHistoryState_(HistoryStates.PAGE_ID) ||
+        firstPageEl.id;
+    const maybePageId = parseQueryString(this.win.location.hash)['page'];
+    const isActualPage =
+      findIndex(this.pages_, page => page.element.id === maybePageId) >= 0;
+    return (maybePageId && isActualPage) ? maybePageId : initialPageId;
```

Comment:
Could we also check that the page ID from the history state exists? This is a bug we're tracking too.

Can we use `getPageIndexById` instead, to avoid code duplication?
I was thinking wrapping these calls in try/catch statements might allow us to use this method (does it?), and still notify the developer that there is something wrong with what they're doing.

Reply:
I think I'm a little confused here. By checking that the page ID from the history state exists, did you mean something more explicit than just `this.getHistoryState_(HistoryStates.PAGE_ID) || firstPageEl.id`?

Also, I don't think we can use a try/catch here because of [how user().error() works](https://github.com/bramanudom/amphtml/blob/master/src/log.js#L261). Just to double check I went ahead and tried, and code in the catch block does not execute despite `getPageIndexById` returning a negative number.

Reply:
Actually, that line of code checks if there is something in the history (`this.getHistoryState_(HistoryStates.PAGE_ID)`), and if not (if that first expression is falsy), it fallbacks the ID of the first page of the story (`firstPageEl.id`).

We're getting a page ID from the history, but just like the page ID from the URL, we can't be sure that the story actually has a page for that ID. Because you just wrote some code to check this, I was wondering if we could also use it to check that the page ID from the history points to an existing page?